tests/integration/test_file_lifecycle.py

```
@@ -323,3 +323,112 @@ def test_file_lifecycle_non_existent_file(monitored_path_with_locations, db_sess
     file_to_disappear.unlink()
+
+
+# ---------------------------------------------------------------------------
+# Operation type migration — end-to-end
+# ---------------------------------------------------------------------------
+
+def test_symlink_to_move_migration(monitored_path_with_locations, db_session):
+    """
+    End-to-end: switch from SYMLINK → MOVE with an already-frozen file.
+
+    Phase 1 – Freeze with SYMLINK:
```

Ruff (0.15.9)

[warning] Docstring contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF002)